src/middleware/github-server-app-middleware.ts

```diff
@@ -2,12 +2,19 @@ import { NextFunction, Request, Response } from "express";
 import { GitHubServerApp } from "models/github-server-app";
 import { Installation } from "models/installation";
 import { envVars } from "config/env";
-import { GITHUB_CLOUD_BASEURL } from "utils/get-github-client-config";
+import { GITHUB_CLOUD_BASEURL, GITHUB_CLOUD_API_BASEURL } from "utils/get-github-client-config";
 
-export const GithubServerAppMiddleware = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+type ResponseType =  Response<
+	{
+    message: string
+	},
+	JiraHostVerifiedLocals
+	& GitHubAppVerifiedLocals
+>;
```

Comment:
Seems like a lot of extra work and gives you a false sense of security since all locals should be optional.  There's no guarantee that anything is set for locals and might cause a runtime error if it's not.

Reply:
Some benefits (from these extra works) I see are:

1. Better typing when we need to use any field from `res.locals`. (And it does actually catch 1 or 2 bugs during this RP).
2. It declares what it relies on (sort of)
3. It might give you false sense of security, but it probably not worse than what it is right now, which is like `just use res.locals.xxx without types whenever you like`. 

---

I agree this PR doesn't address the security feeling/promise of the codebase, I'd prefer a more proper routing setup somehow to guarantee that.